tests/rules/no-missing-atx-heading-space.test.js

```diff
@@ -0,0 +1,423 @@
+	{
+		code: dedent`# Heading 1
+
+		##Heading 2
+		
+		### Heading 3`,
+		output: dedent`# Heading 1
+
+		## Heading 2
+		
+		### Heading 3`,
+		errors: [
+			{
+				messageId: "missingSpace",
+				line: 3,
+				column: 2,
+				endLine: 3,
+				endColumn: 11,
+			},
+		],
+	},
+
+	// 3. Simple heading variations
+	{
+		code: "#Text",
+		output: "# Text",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 5,
+			},
+		],
+	},
+	{
+		code: "#Heading with trailing space ",
+		output: "# Heading with trailing space ",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 28,
+			},
+		],
+	},
+
+	// 4. Headings containing code-like syntax
+	// 4.1 With fenced code markers
+	{
+		code: "#Something with ``` backticks",
+		output: "# Something with ``` backticks",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 29,
+			},
+		],
+	},
+	// 4.2 With backticks mid-heading
+	{
+		code: "#Heading with ``` in the middle and more text after",
+		output: "# Heading with ``` in the middle and more text after",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 51,
+			},
+		],
+	},
+	// 4.3 With inline code
+	{
+		code: "#Heading with `inline code`",
+		output: "# Heading with `inline code`",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 14,
+			},
+		],
+	},
+	// 4.4 With tilde markers
+	{
+		code: "#Title with ~~~ tildes in it",
+		output: "# Title with ~~~ tildes in it",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
+				endColumn: 28,
+			},
+		],
+	},
+
+	// 5. Multi-line documents
+	// 5.1 With code markers in context
+	{
+		code: dedent`Text before
+		#Heading with \`\`\` code markers
+		Text after`,
+		output: dedent`Text before
+		# Heading with \`\`\` code markers
+		Text after`,
+		errors: [
+			{
+				messageId: "missingSpace",
+				line: 2,
+				column: 1,
+				endLine: 2,
+				endColumn: 30,
+			},
+		],
+	},
+	// 5.2 Multiple incorrect headings in one file
+	{
+		code: dedent`#First heading
+		
+		Some text
+		
+		##Second heading
+		
+		###Third heading`,
+		output: dedent`# First heading
+		
+		Some text
+		
+		## Second heading
+		
+		### Third heading`,
+		errors: [
+			{
+				messageId: "missingSpace",
+				line: 1,
+				column: 1,
+				endLine: 1,
+				endColumn: 14,
+			},
+			{
+				messageId: "missingSpace",
+				line: 5,
+				column: 2,
+				endLine: 5,
+				endColumn: 16,
+			},
+			{
+				messageId: "missingSpace",
+				line: 7,
+				column: 3,
+				endLine: 7,
+				endColumn: 16,
+			},
+		],
+	},
+
+	// 6. Emphasis handling
+	{
+		code: "#*hi*",
+		output: "# *hi*",
+		errors: [
+			{
+				messageId: "missingSpace",
+				column: 1,
+				line: 1,
+				endLine: 1,
```

Comment:
```suggestion
				endLine: 6,
```

It should be the length of `#*hi*`, since other errors follow the same error location format.

The same applies to the other Emphasis handling test cases.

Reply:
Fixed the endLine as per https://github.com/eslint/markdown/pull/371#discussion_r2098403409